### ISSUE TITLE
High: Issues with shipping prices

### ISSUE THREAD

**Issue by Xyand** [printdotio/printio-ios-sdk#350]
When trying to order pillows to a New Jersey address I see a shipping price of around $26 for standard shipping. I inspected the request and found that a different shipping price is returned when state is formatted differently. I used the SDK to auto-correct the address.

Request issued by the SDK:

````
{
	"ShipToState": "NEW JERSEY",
	"ShipToPostalCode": "07936",
	"CurrencyCode": "USD",
	"ShipToCountry": "US",
	"LanguageCode": "en",
	"Items": [{
		"SKU": "ThrowPillow_Zipper_16x16",
		"Quantity": 1
	}]
}
````

Response:
````
{
	"Result": [{
		"SKUs": ["ThrowPillow_Zipper_16x16"],
		"ShipOptions": [{
			"CarrierName": "Standard",
			"MethodType": "Standard",
			"Name": "Standard",
			"Price": {
				"Price": 26.31,
				"CurrencyCode": "USD",
				"FormattedPrice": "$26.31",
				"CurrencyFormat": "${1}",
				"CurrencyDigits": 2
			},
			"Id": 1,
			"EstBusinessDaysTilDelivery": 12
		}]
	}]
}
````

I can actually change the state to any bogus string and get the same result.

When I modified the address to `NJ` much more realistic prices were returned. I wonder if this isn't a sign of a larger issue.

Modified request:
````
{
	"ShipToState": "NJ",
	"ShipToPostalCode": "07936",
	"CurrencyCode": "USD",
	"ShipToCountry": "US",
	"LanguageCode": "en",
	"Items": [{
		"SKU": "ThrowPillow_Zipper_16x16",
		"Quantity": 1
	}]
}
````

Response:
````
{
    "Result": [
        {
            "SKUs": [
                "ThrowPillow_Zipper_16x16"
            ],
            "ShipOptions": [
                {
                    "CarrierName": "Standard",
                    "MethodType": "Standard",
                    "Name": "Standard",
                    "Price": {
                        "Price": 7.03,
                        "CurrencyCode": "USD",
                        "FormattedPrice": "$7.03",
                        "CurrencyFormat": "${1}",
                        "CurrencyDigits": 2
                    },
                    "Id": 1,
                    "EstBusinessDaysTilDelivery": 12
                },
                {
                    "CarrierName": "Overnight",
                    "MethodType": "Overnight",
                    "Name": "Overnight",
                    "Price": {
                        "Price": 26.12,
                        "CurrencyCode": "USD",
                        "FormattedPrice": "$26.12",
                        "CurrencyFormat": "${1}",
                        "CurrencyDigits": 2
                    },
                    "Id": 3,
                    "EstBusinessDaysTilDelivery": 2
                }
            ]
        }
    ]
}
````

**Comment by perisicboro** [comment]
@Xyand When US is selected country, 'state' field is not editable. How did you menage to get this: ```"ShipToState": "NEW JERSEY"``` ?

**Comment by Xyand** [comment]
I entered the state and then when the system wanted to correct the address (next screen) I agreed to the corrected address.

**Comment by AustenB** [comment]
Typically it should correct the address to a two digit code.

Interesting use case. I guess we passed it new jersey and it didn't verify

**Comment by Xyand** [comment]
Could you look into it? It may be something broader that affect shipping prices -> conversions.

**Comment by AustenB** [comment]
Can you get this to reoccur?

On Wed, Jan 21, 2015 at 4:12 PM, Albert <notifications@github.com> wrote:

> Could you look into it? It may be something broader that affect shipping
> prices -> conversions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/printdotio/printio-ios-sdk/issues/350#issuecomment-70922568>
> .
>



-- 
Austen Bernstein
Founder & CEO
printio

**Comment by perisicboro** [comment]
@Xyand It should be fixed in 1.0.41-dev (will be this evening)

**Comment by AustenB** [comment]
@Xyand confirmed
